File: competitions/Week_of_Code_35/3d_surface_area.py
#https://www.hackerrank.com/contests/w35/challenges/3d-surface-area

def solution(table, h, w):
    area = h * w * 2
    for i in range(h):
        for j in range(w):
            height = table[i][j]
            if i == 0:
                area += height
            if j == 0:
                area += height
            if i == h-1:
                area += height
            if j == w-1:
                area += height
            if i != 0:
                area += max(height - table[i-1][j],0)
            if j != 0:
                area += max(height - table[i][j-1],0)
            if i != h-1:
                area += max(height - table[i+1][j],0)
            if j != w-1:
                area += max(height - table[i][j+1],0)
    return area
    

if __name__ == "__main__":
    h, w = [int(_) for _ in input().strip().split()]
    table = []
    for i in range(h):
        row = [int(_) for _ in input().strip().split()]
        table.append(row)
    ans = solution(table, h, w)
    print(ans)

File: competitions/Week_of_Code_35/3d_surface_area.cpp
#include <bits/stdc++.h>

using namespace std;

int main() {
	int h, w;
	cin >> h >> w;
	int table[h][w];
	for (int i = 0; i < h; i++) {
		for (int j = 0; j < w; j++) {
			int val;
			cin >> val;
			table[i][j] = val;
		}
	}

	int area = 2 * w * h;
	for (int i = 0; i < h; i++) {
		for (int j = 0; j < w; j++) {
			if (i == 0) {
				area += table[i][j];
			}
			if (j == 0) {
				area += table[i][j];
			}
			if (i == h - 1) {
				area += table[i][j];
			}
			if (j == w - 1) {
				area += table[i][j];
			}
			if (i != 0) {
				area += max(0, table[i][j] - table[i-1][j]);
			}
			if (j != 0) {
				area += max(0, table[i][j] - table[i][j-1]);
			}
			if (i != h-1) {
				area += max(0, table[i][j] - table[i+1][j]);
			}
			if (j != w-1) {
				area += max(0, table[i][j] - table[i][j+1]);
			}
		}
	}
	cout << area;

	return 0;
}
